hack/job.sh

```diff
@@ -10,10 +10,16 @@ if [[ -n "${GITHUB_TOKEN_PATH:-}" ]]; then
 fi
 
 CONTAINER_ENGINE=${CONTAINER_ENGINE:-docker}
+VOLUME_MOUNT_FLAGS=${VOLUME_MOUNT_FLAGS:-:z}
+
+if [ "$CONTAINER_ENGINE" == "podman" ] && [ "$(uname -s)" == "Darwin" ]; then
```

Comment:
I'm not a mac user so I could be wrong but shouldn't `CONTAINER_ENGINE=podman JOB=xyz make job` already handle this scenario? The following piece of code seems to be the same: https://github.com/openshift/release/blob/3b5302c778c57b679c361e44eea914dcbc6a2c73/Makefile#L9-L18

Reply:
Good catch!
Let me revise it.

Reply:
Sorry. I was thinking something else.
The vars referred in your comment are for the podman cmd in the Makefile.
This PR applied the same trick to the `hack/job.sh` script to handle the same problem which is`:z`  does not work on Mac.

Reply:
The change `--volume "$PWD:/tmp/release${VOLUME_MOUNT_FLAGS}" \`  you have introduced looks fine to me, I understand what you mean. On the other hand [this](https://github.com/hongkailiu/release/blob/337141cc073662fa9e342ae24c9faf6d873ceaa6/hack/job.sh#L15-L18) seems to be redundant due to [that](https://github.com/hongkailiu/release/blob/337141cc073662fa9e342ae24c9faf6d873ceaa6/Makefile#L9-L18).
If you use `make` to run `hack/job.sh` like so `CONTAINER_ENGINE=podman JOB=xyz make job` I expect `VOLUME_MOUNT_FLAGS` to be empty on Mac, hence I would keep `--volume "$PWD:/tmp/release${VOLUME_MOUNT_FLAGS}" \` only and remove the [VOLUME_MOUNT_FLAGS check](https://github.com/hongkailiu/release/blob/337141cc073662fa9e342ae24c9faf6d873ceaa6/hack/job.sh#L15-L18).  Does it make sense?

Reply:
I tried to run it after removing the code you suggested. It did not work.
The reason is that when we invoke `job.sh` in the Makefile, we did not passed the env. vars onto it.
Actually, we could do `VOLUME_MOUNT_FLAGS=$(VOLUME_MOUNT_FLAGS) hack/job.sh "$(JOB)"` instead.
It did not work out either, because `VOLUME_MOUNT_FLAGS` is an empty string and `VOLUME_MOUNT_FLAGS=${VOLUME_MOUNT_FLAGS:-:z}` did not distinguish if it is unset or empty.

`if [ -z ${VOLUME_MOUNT_FLAGS+x} ]; then VOLUME_MOUNT_FLAGS=':z'; else echo "VOLUME_MOUNT_FLAGS is set to '$VOLUME_MOUNT_FLAGS'"; fi` seems does the job.